apps/framework-cli/src/framework/aggregations.ts

```diff
@@ -0,0 +1,164 @@
+    `Connecting to Clickhouse at ${protocol}://${CLICKHOUSE_HOST}:${CLICKHOUSE_PORT}`,
+  );
+  return createClient({
+    url: `${protocol}://${CLICKHOUSE_HOST}:${CLICKHOUSE_PORT}`,
+    username: CLICKHOUSE_USERNAME,
+    password: CLICKHOUSE_PASSWORD,
+    database: CLICKHOUSE_DB,
+  });
+};
+
+const waitForClickhouse = (chClient: ClickHouseClient) => {
+  const pingClickhouse = async () => {
+    try {
+      const queryRespose = await chClient.query({ query: "SHOW TABLES" });
+      const showTablesResponse =
+        (await queryRespose.json()) as ShowTablesResponse;
+      return showTablesResponse.rows > 0;
+    } catch (_) {
+      return false;
+    }
+  };
+
+  const poll = (
+    resolve: (value?: unknown) => void,
+    reject: (reason?: any) => void,
+  ) => {
+    pingClickhouse().then((condition) => {
+      if (condition) resolve();
+      else setTimeout(() => poll(resolve, reject), 1000);
+    });
+  };
+
+  return new Promise(poll);
+};
+
+const getVersion = () => {
+  const version = JSON.parse(Deno.readTextFileSync(`${cwd}/package.json`))
+    .version as string;
+  return version.replace(/\./g, "_");
+};
+
+const getFileName = (filePath: string) => {
+  const regex = /\/([^\/]+)\.ts/;
+  const matches = filePath.match(regex);
+  if (matches && matches.length > 1) {
+    return matches[1];
+  }
+  return "";
+};
+
+const createAggregation = async (
+  chClient: ClickHouseClient,
+  path: string,
+  version: string,
+) => {
+  const fileName = getFileName(path);
+
+  try {
+    const sqlString = (await import(path)).default;
+    if (typeof sqlString !== "string") {
+      console.error(
+        `Not creating aggregation. Expected an export default SQL string from ${fileName}`,
+      );
+      return;
+    }
+
+    const mvQuery = `
+          CREATE MATERIALIZED VIEW IF NOT EXISTS ${fileName}Mv
+          TO ${fileName}_${version}
+          AS ${sqlString}
+      `;
+    await chClient.command({ query: mvQuery });
+    console.log(`Created aggregation ${fileName}`);
+  } catch (err) {
+    console.error(`Failed to create aggregation ${fileName}: ${err}`);
+  }
+};
+
+const deleteAggregation = async (
+  chClient: ClickHouseClient,
+  path: string,
+  version: string,
+) => {
+  const fileName = getFileName(path);
+
+  try {
+    await chClient.command({ query: `DROP VIEW IF EXISTS ${fileName}Mv` });
+    await chClient.command({
+      query: `DROP TABLE IF EXISTS ${fileName}_${version}`,
+    });
+    console.log(`Deleted aggregation ${fileName}`);
+  } catch (err) {
+    console.error(`Failed to delete aggregation ${fileName}: ${err}`);
+  }
+};
+
+const startFileWatcher = (chClient: ClickHouseClient) => {
+  const version = getVersion();
+  const pathToWatch = `${AGGREGATIONS_DIR_PATH}/**/${AGGREGATIONS_FILE}`;
+
+  watch(pathToWatch, { usePolling: true }).on(
+    "all",
+    async (event: string, path: string) => {
+      const antiCachePath = `${path}?num=${Math.random().toString()}&time=${Date.now()}`;
```

Comment:
User could write an aggregation then we create a materialized view in clickhouse. If they change the sql string, we get a "change" event but it's going to import the same file path so it just uses what's already cached. Appending url params is a workaround: `https://github.com/denoland/deno/issues/13754#issuecomment-1908241494`